internal/otelcollector/config/metric/agent/config_builder.go

```diff
@@ -128,3 +187,15 @@ func makePipelinesConfig(inputs inputSources) config.Pipelines {
 
 	return pipelinesConfig
 }
+
+func makeRuntimePipelineProcessorsIDs(runtimeResources runtimeResourcesEnabled) []string {
+	processors := []string{"memory_limiter"}
+
+	if runtimeResources.volume {
+		processors = append(processors, "filter/drop-non-pvc-volumes-metrics")
+	}
+
+	processors = append(processors, "resource/delete-service-name", "transform/set-instrumentation-scope-runtime", "transform/insert-skip-enrichment-attribute", "batch")
```

Comment:
is the order important? If not it would imho read better like this:
```suggestion
	processors := []string{"memory_limiter"
	"resource/delete-service-name", 
	"transform/set-instrumentation-scope-runtime", 
	"transform/insert-skip-enrichment-attribute", 
	"batch",
	}

	if runtimeResources.volume {
		processors = append(processors, "filter/drop-non-pvc-volumes-metrics")
	}
```

Reply:
It is better in terms of performance to drop the unwanted metrics in the beginning to avoid doing all the other processing for metrics which will be dropped at the end.

But I'm not strongly against doing your suggestion for better readability. What do you think?

Reply:
if it impacts performance then please keep it as is!